### PR TITLE
fix: subsetLoadBalancer.HostNum do not consider fallback policy

### PR DESCRIPTION
**Pull request by antJack**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Review by taoyuanyuan — DISMISSED** [review]
Lgtm

File: pkg/upstream/cluster/subset_loadbalancer.go [hunk 1]
```diff
@@ -85,6 +85,9 @@ func (sslb *subsetLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) int
 		matchCriteria := metadata.MetadataMatchCriteria()
 		entry := sslb.findSubset(matchCriteria)
 		if entry == nil {
+			if sslb.fallbackSubset != nil {
```
Comment (nejisama) [inline pkg/upstream/cluster/subset_loadbalancer.go:88]: host number增加了兜底判断以后，IsExistsHost 要不要加呢？

Comment (antJack) [reply pkg/upstream/cluster/subset_loadbalancer.go:91]: fix

Comment (taoyuanyuan) [reply pkg/upstream/cluster/subset_loadbalancer.go:91]: 需要要跟choosehost结果一样， !entry.Active()也是需要判断的

File: pkg/upstream/cluster/subset_loadbalancer.go [hunk 2]
```diff
@@ -84,7 +90,13 @@ func (sslb *subsetLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) int
...
 		matchCriteria := metadata.MetadataMatchCriteria()
 		entry := sslb.findSubset(matchCriteria)
-		if entry == nil {
+		if entry == nil && sslb.fallbackSubset != nil {
+			if log.DefaultLogger.GetLogLevel() >= log.DEBUG {
+				log.DefaultLogger.Debugf("[upstream] [subset lb] HostNum entry is nil, do fallback")
+			}
+			return sslb.fallbackSubset.LoadBalancer().HostNum(metadata)
+		}
+		if entry == nil || !entry.Active() {
```
Comment (taoyuanyuan) [inline pkg/upstream/cluster/subset_loadbalancer.go:99]: 看choosehost，在!entry.Active()的场景下，也是需要走fallback的哟

**Review by taoyuanyuan — DISMISSED** [review]
LGTM

**Review by nejisama — APPROVED** [review]

**Review by taoyuanyuan — APPROVED** [review]
lgtm

**Pull request merged**